docs/cryptodoc/src/05_10_cmce.rst

```diff
@@ -0,0 +1,469 @@
+to avoid early abortion if any check fails.
+
+
+.. _pubkey/cmce/params:
+
+Parameters
+^^^^^^^^^^
+
+The ``Classic_McEliece_Parameter_Set`` enum contains all instances of the
+Classic McEliece algorithm listed in Table :ref:`Supported Classic McEliece
+Parameter Sets <pubkey/cmce/parameter_table>`. These parameter sets serve as
+the basis for deriving all the necessary parameters used within the algorithm.
+They are collected in a ``Classic_McElice_Parameters`` object, which includes
+all the parameters defined in the specifications [CMCE-ISO]_ and [CMCE-R4]_.
+This object is passed to all algorithm components, ensuring consistent
+parameter usage.
+
+
+.. _pubkey/cmce/key_gen:
+
+
+Key Generation
+--------------
+
+Classic McEliece key generation follows Section 8.3 of [CMCE-ISO]_ and is
+implemented within the ``Classic_McEliece_KeyPair_Internal`` class. It works as
+follows:
+
+.. admonition:: Classic McEliece Key Generation
+
+   **Input:**
+
+   -  ``rng``: random number generator
+
+   **Output:**
+
+   -  ``SK``, ``PK``: private and public key
+
+   **Steps:**
+
+   1. Generate a random value ``seed`` using ``rng``
+   2. ``s, ordering_bits, irreducible_bits, next_seed = PRF(seed)``
+   3. | Create a field ordering ``field_ordering`` using ``ordering_bits``
+      | On failure, set ``seed = next_seed`` and go to step 2
+
+   4. | Create a monic irreducible polynomial ``g`` using ``irreducible_bits``
+      | On failure, set ``seed = next_seed`` and go to step 2
+
+   5. | Create a parity check matrix in systematic form ``H = (I_mt | T)`` using ``field_ordering`` and ``g``. Meanwhile, compute the column selection ``c``
+      | On failure, set ``seed = next_seed`` and go to step 2
+
+   6. ``SK = {seed, c, g, field_ordering, s}, PK = {T}``
+
+   **Notes:**
+
+   - ``PRG`` is an application of ``SHAKE256`` with an input prefix byte 64.
+     The output length is ``n/8 + 4q + 2t + 32`` bytes. It is defined
+     in Section 9.1 of [CMCE-ISO]_.
+   - Only the first ``n`` elements of the field ordering are used to
+     create the parity check matrix.
+   - For instances with the suffix ``f``, the semi-systematic transformation
+     strategy is employed. In this case, ``field_ordering`` is updated to
+     reflect the column selection.
+   - To store the private and public keys as bytes, the respective entries are
+     serialized following the specifications of Section 9.2 of [CMCE-ISO]_.
+
+
+.. _pubkey/cmce/encapsulation:
+
+Key Encapsulation
+-----------------
+
+The Classic McEliece encapsulation procedure of Botan follows Section 8.5 of
+[CMCE-ISO]_ and works as follows:
+
+.. admonition:: Classic McEliece Encapsulation
+
+   **Input:**
+
+   - ``PK = {T}``: public key
+   - ``rng``: random number generator
+
+   **Output:**
+
+   - ``encap_key``: ciphertext of shared key
+   - ``shared_key``: plaintext shared key
+
+   **Steps:**
+
+   1. Generate a random error vector ``e`` of weight ``t`` using ``rng``
```

Comment:
This is fine but I'd be more of a fan to write:

`2. ``H = Classic_McEliece_Matrix`` representing I_mt | T, i.e., to get the previously described names and classes into the pseudocode.